streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/pe/DataStreamResource.java

```diff
@@ -42,7 +49,17 @@
 
 @RestController
 @RequestMapping("/api/v2/streams")
-public class DataStreamResource extends AbstractAuthGuardedRestResource {
+public class DataStreamResource extends AbstractAdapterResource<AdapterMasterManagement> {
+
+  public DataStreamResource() {
+    super(() -> new AdapterMasterManagement(
```

Comment:
Can you tell me what the reason is why you added this code? I do not completely understand it.

Reply:
I needed access to AdapterMasterManagement instance. I used the same pattern as in AdapterResource to bring it in